samples/drivers/flash_shell/src/main.c

```diff
@@ -227,6 +248,143 @@ static int do_write(const struct shell *shell, off_t offset, uint8_t *buf,
 	return ret;
 }
 
+static int do_write_unaligned(const struct shell *shell, off_t offset, uint8_t *buf,
+		    size_t len, bool read_back)
+{
+	int ret;
+	size_t page_size = flash_get_write_block_size(flash_device);
+	size_t size_before = offset % page_size;
+	size_t size_after = page_size - ((size_before + len) % page_size);
+	size_t aligned_size = size_before + len + size_after;
+	off_t  start_page = offset - size_before;
+	off_t  last_page = start_page + aligned_size - page_size;
+	bool single_page_write = (size_before + len < page_size);
+
+	static char *before_data;
```

Comment:
Any reason these are static? Leftover from when they were buffers in RAM?

Reply:
You are right it's a leftover from when allocating buffers once.